### ISSUE TITLE
the table name with the underline exception occurred when `show table` is performed

### ISSUE THREAD

**Issue by 80631004** [pingcap/tidb#5935]
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
1）：创建表
CREATE TABLE person_1_2_3 (
    number INT(11),
    name VARCHAR(255),
    birthday DATE
    );

2）执行show tables;
显示表名为：person_
2. What did you expect to see?
执行show tables;
希望为显示为创建的表名:
person_1_2_3

3. What did you see instead?
显示表名为：person_

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
Release Version: v1.1.0-beta-17-g816c764
Git Commit Hash: 816c764cb5d1a144f441f269782b51a23f97db9e
Git Branch: master
UTC Build Time: 2018-02-27 03:07:20
GoVersion: go version go1.10 linux/amd64

**Comment by zimulala** [comment]
PTAL @jackysp

**Comment by jackysp** [comment]
Thank you for your feedback, @80631004 , but I can not reproduce your issue.
```
mysql> CREATE TABLE person_1_2_3 (
    -> number INT(11),
    -> name VARCHAR(255),
    -> birthday DATE
    -> );
Query OK, 0 rows affected (0.20 sec)

mysql> show tables;
+----------------+
| Tables_in_test |
+----------------+
| person_1_2_3   |
| t              |
+----------------+
2 rows in set (0.00 sec)

mysql> select tidb_version();
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                     |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v1.1.0-beta-17-g816c764c
Git Commit Hash: 816c764cb5d1a144f441f269782b51a23f97db9e
Git Branch: HEAD
UTC Build Time: 2018-03-02 02:29:36
GoVersion: go version go1.10 darwin/amd64 |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

**Comment by jackysp** [comment]
I've tried it on Linux, cannot reproduce either. Could you please show more details for your test, @80631004  ?

**Comment by 80631004** [comment]
@jackysp  I am use  Navicat Premium tools  to create table and show table

**Comment by jackysp** [comment]
So `show tables` results are correct if you use MySQL client? @80631004

**Comment by 80631004** [comment]
not sure,tonigth I will try it

**Comment by jackysp** [comment]
The table name looks good on Navicat Premium 12.0.20.

**Comment by 80631004** [comment]
@jackysp   use mysql client to show table is correct, I am use Navicat Premium 11.2.7 tools to create table, So Navicat Premium 11.2.7 sames have some wrong for show table

**Comment by shenli** [comment]
@80631004 Which OS are you using?

**Comment by 80631004** [comment]
windows 7

**Comment by jackysp** [comment]
I can reproduce this bug. TiDB set a default field length for all show schemas at https://github.com/pingcap/tidb/blob/4e1a114e68e757a68b6b2677b0fde6228b090892/plan/planbuilder.go#L1262 ,
which is not compatible with MySQL.

TiDB:

```
mysql> show tables;
Field   1:  `Tables_in_test`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       VAR_STRING
Collation:  utf8_general_ci (33)
Length:     20
Max_length: 9
Decimals:   0
Flags:      


+----------------+
| Tables_in_test |
+----------------+
| VOTES          |
| followers      |
| nation         |
| region         |
| supplier       |
| t              |
+----------------+
6 rows in set (0.00 sec)
```

and MySQL:

```
mysql> show tables;
Field   1:  `Tables_in_test`
Catalog:    `def`
Database:   `information_schema`
Table:      `TABLE_NAMES`
Org_table:  `TABLE_NAMES`
Type:       VAR_STRING
Collation:  utf8_general_ci (33)
Length:     192
Max_length: 14
Decimals:   0
Flags:      NOT_NULL 


+----------------+
| Tables_in_test |
+----------------+
| SHOW_test      |
| VOTES          |
| abc            |
| abcdefghijklmn |
| followers      |
| follows        |
| hb             |
| notnull        |
| t              |
| t1             |
| t-t-t-t1       |
| ta             |
| test           |
| test_defnull   |
| tidb_test      |
| tt             |
| ttt            |
| tttt           |
| user_profiles  |
+----------------+
19 rows in set (0.00 sec)
```

TiDB caculate an accurate `Max_length` of field type as MySQL, but the `Length` is different. I think navicat 11 use the `Length` field for show the sql results.

**Comment by jackysp** [comment]
@80631004 , this bug was fixed by #5945 . You can use the latest master branch. Thank you for your report.
